components/provisioner/internal/credentials/provider.go

```diff
@@ -0,0 +1,38 @@
+package credentials
+
+type TargetProvider string
+
+const (
+	GCP   TargetProvider = "GCP"
+	Azure TargetProvider = "Azure"
+	AWS   TargetProvider = "AWS"
+)
+
+type Kubeconfig []byte
+
+type GardenerCredentials struct {
+	ClusterCredentials Kubeconfig
+	SecretName         string
```

Comment:
Shouldn't we use CredentialsSecretName instead, to be consistent with current naming? Maybe these Credentials are sth different from those required for provisioning?

```suggestion
	CredentialsSecretName         string
```

Reply:
All clear :)